### PR TITLE
gh-113299: Create libclinic package

### PR DESCRIPTION
**Pull request by vstinner**
* Create Tools/clinic/libclinic/ package.
* Move cpp.py to libclinic.
* Create libclinic.utils.

* Issue: gh-113299

### CODE REVIEW

**Comment by vstinner** [comment]
@erlend-aasland @AlexWaygood @corona10: Ok, here is first minimum step to create a new package for Argument Clinic code.

File: Tools/clinic/libclinic/utils.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    overload,
+    TYPE_CHECKING,
+)
+
+if TYPE_CHECKING:
+    from clinic import Clinic
+
+
+# Clinic instance
+clinic: 'Clinic' | None = None
```
Comment (AlexWaygood) [inline Tools/clinic/libclinic/utils.py:16]: This PR feels like it _only_ does my least favourite thing about your other PR :// https://github.com/python/cpython/issues/113299#issuecomment-1864092370

File: Tools/clinic/clinic.py [hunk 2]
```diff
@@ -2681,6 +2613,7 @@ def __init__(
 
         global clinic
         clinic = self
+        utils.clinic = self
```
Comment (erlend-aasland) [inline Tools/clinic/clinic.py:2616]: IMO, putting a global from one file into another file is not a good idea.

**Comment by vstinner** [comment]
@erlend-aasland:
> IMO, putting a global from one file into another file is not a good idea.

I don't have the bandwidth right now to move this global variable. Adding `utils.clinic` variable is a temporary solution until someone can spend time to work on it. I gave on my latest attempt to partially fix the issue since nobody was available to review my work: https://github.com/python/cpython/pull/110984

Apparently, it's a requirement for @AlexWaygood and @erlend-aasland so I prefer to close this PR for now.

**Pull request closed without merging**